### ISSUE TITLE
just installing and using plots seems to make it git dirty

### ISSUE THREAD

**Issue by lewisl** [tbreloff/Plots.jl#55]
This is really bad.  With an unstaged "change", you can't use any Pkg commands on the package because git thinks it is dirty.  I didn't "do" anything except using Plots, immerse(), plot(1:4).  I did the same thing with pyplot().

This is very painful for users to fix, requiring much mucking about with git.  Standard things like git checkout -- .  do NOT work to fix this.  I ended up rm'ing the file and rebasing the repository, then Pkg.rm'ing it to get rid of it and then reinstalling.

And even removing things leaves lots of detritus just in case you want to reinstall-so you never really get it clean unless you delete from the file system.  and then you've got out of sync caches.

Here is the outcome:

shell> cd ..
/Users/lewislevinmbr/.julia/v0.4/Plots
shell> git status
HEAD detached at 894928d
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   img/qwt/qwt_example_2.gif

no changes added to commit (use "git add" and/or "git commit -a")

**Comment by tbreloff** [comment]
Can you give more details? What does this give:

```
cd ~/.julia/v0.4/Plots
git status
```

**Comment by tbreloff** [comment]
Also, can you give me an idea of what you did?  I assume you were changing the notebooks in the examples directory?

**Comment by tbreloff** [comment]
FYI... I had already started to transition these things to https://github.com/tbreloff/ExamplePlots.jl.  Of course it doesn't change the fact that you'd then make the ExamplePlots repo dirty if you change the examples... you should copy/download the examples to another directory to avoid that.

**Comment by lewisl** [comment]
Here you go.  I deleted my ~/.julia/0.4.0 packages directory.  Re-installed packages.

julia> using Plots

julia> pyplot()
Plots.PyPlotPackage()

julia> plot(1:4)
[Plots.jl] Initializing backend: pyplot
PyPlot.Figure(PyObject <matplotlib.figure.Figure object at 0x3193dab90>)

julia> Pkg.status()
4 required packages:
 - IJulia                        1.1.7
 - JLD                           0.5.5
 - Plots                         0.4.1              894928d7 (dirty)
 - PyPlot                        2.1.1
19 additional packages:
 - BinDeps                       0.3.18
 - Blosc                         0.1.4
 - ColorTypes                    0.1.7
 - Colors                        0.5.4
 - Compat                        0.7.7
 - Conda                         0.1.7
 - Dates                         0.4.4
 - Docile                        0.5.19
 - FixedPointNumbers             0.0.12
 - HDF5                          0.5.6
 - Homebrew                      0.1.17
 - JSON                          0.5.0
 - LaTeXStrings                  0.1.6
 - Nettle                        0.2.0
 - PyCall                        1.1.2
 - Reexport                      0.0.3
 - SHA                           0.1.2
 - URIParser                     0.1.1
 - ZMQ                           0.3.0


So, all I did was a plot.

Here is the git output:

shell> cd ~/.julia
/Users/lewislevinmbr/.julia

shell> ls
lib v0.3 v0.4

shell> cd v0.4
/Users/lewislevinmbr/.julia/v0.4

shell> cd Plots
/Users/lewislevinmbr/.julia/v0.4/Plots

shell> git status
HEAD detached at 894928d
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

modified:   examples/meetup/iheartplots.gif
modified:   img/gadfly/gadfly_example_2.gif
modified:   img/immerse/immerse_example_2.gif
modified:   img/pyplot/pyplot_example_2.gif
modified:   img/qwt/qwt_example_2.gif

no changes added to commit (use "git add" and/or "git commit -a")






On Oct 27, 2015, at 10:36 AM, Tom Breloff <notifications@github.com<mailto:notifications@github.com>> wrote:


Also, can you give me an idea of what you did? I assume you were changing the notebooks in the examples directory?

**Comment by lewisl** [comment]
But, you can’t:  you get the examples when you Pkg.add(“Plots”).

I sent Tom the output of git status.

You guys are great.

To be fair, there is an uphill challenge to be better than PyPlot.  Still a more “pure” (whatever that means!) julia plotting solution with native backends for Windows and Mac and the “right” “popular” gui libraries on ‘nix would be nice, but lots’o’work.


On Oct 27, 2015, at 10:48 AM, Tom Breloff <notifications@github.com<mailto:notifications@github.com>> wrote:


FYI... I had already started to transition these things to https://github.com/tbreloff/ExamplePlots.jl. Of course it doesn't change the fact that you'd then make the ExamplePlots repo dirty if you change the examples... you should copy/download the examples to another directory to avoid that.

**Comment by tbreloff** [comment]
There's something fishy about this output.  I think you didn't actually delete your package directory, but it's hard to know exactly what you did.  If you deleted the Plots directory (as in, completely removed it from your filesystem), and then:

```
Pkg.add("Plots")
using Plots
plot(rand(10))
```

...I would expect the package to be clean.

Also, the examples and img directories are (mostly) gone on Plots master now.  You can follow that with a `Pkg.checkout("Plots")`

Finally, if you ever get a dirty package, you can clean it up with: `git reset --hard`, with the obvious warning that you'll lose any changes you made in that directory.

Let me know if you have any other questions/issues.

**Comment by Ken-B** [comment]
I got the same error. Note that I have never changed anything in the package or even ran the examples. I just used it with `using Plots`.

Specifically, running `Pkg.update()` yields
```
ulia> Pkg.update()
INFO: Updating METADATA...
INFO: Updating cache of Images...
INFO: Updating cache of ColorTypes...
INFO: Updating cache of Plots...
INFO: Updating cache of Requires...
INFO: Updating cache of Gadfly...
INFO: Updating cache of JLD...
INFO: Updating cache of Memoize...
INFO: Updating cache of Logging...
INFO: Updating cache of LightGraphs...
INFO: Updating cache of NullableArrays...
INFO: Updating cache of ColorVectorSpace...
INFO: Updating cache of FixedPointNumbers...
INFO: Updating cache of FileIO...
INFO: Updating cache of Homebrew...
INFO: Updating cache of StatsFuns...
INFO: Updating cache of TexExtensions...
INFO: Updating cache of Colors...
INFO: Updating cache of SIUnits...
INFO: Updating cache of ImageMagick...
INFO: Updating cache of Calculus...
INFO: Updating Rutherford...
INFO: Updating ForwardDiff...
INFO: Updating Atom...
INFO: Updating MinFinder...
INFO: Updating JuliaParser...
INFO: Updating SymPy...
INFO: Updating Compose...
INFO: Updating ImageIO...
INFO: Updating ImageMagick...
INFO: Updating Gadfly...
INFO: Updating FileIO...
INFO: Updating Benchmarks...
INFO: Updating Electron...
INFO: Updating Sobol...
INFO: Updating LeafAreaIndex...
INFO: Updating Jewel...
INFO: Updating Nettle...
INFO: Computing changes...
INFO: Cloning cache of Clustering from git://github.com/JuliaStats/Clustering.jl.git
INFO: Upgrading Calculus: v0.1.13 => v0.1.14
INFO: Installing Clustering v0.5.0
INFO: Upgrading ColorTypes: v0.1.7 => v0.2.0
INFO: Upgrading ColorVectorSpace: v0.0.5 => v0.1.0
INFO: Upgrading Colors: v0.5.4 => v0.6.0
INFO: Upgrading FixedPointNumbers: v0.0.12 => v0.1.0
INFO: Upgrading Homebrew: v0.1.16 => v0.1.17
INFO: Upgrading Images: v0.4.50 => v0.5.0
INFO: Upgrading JLD: v0.5.5 => v0.5.6
INFO: Upgrading LightGraphs: v0.3.4 => v0.4.0
INFO: Upgrading Logging: v0.1.1 => v0.2.0
INFO: Upgrading Memoize: v0.0.0 => v0.0.1
INFO: Upgrading NullableArrays: v0.0.1 => v0.0.2
INFO: Upgrading Plots: v0.4.0 => v0.4.1
git-lfs smudge 'examples/meetup/iheartplots.gif': git-lfs: command not found
error: cannot feed the input to external filter git-lfs smudge %f
error: external filter git-lfs smudge %f failed -1
error: external filter git-lfs smudge %f failed
fatal: examples/meetup/iheartplots.gif: smudge filter lfs failed
INFO: Rolling back NullableArrays from v0.0.2 to v0.0.1
INFO: Rolling back Memoize from v0.0.1 to v0.0.0
INFO: Rolling back Logging from v0.2.0 to v0.1.1
INFO: Rolling back LightGraphs from v0.4.0 to v0.3.4
INFO: Rolling back JLD from v0.5.6 to v0.5.5
INFO: Rolling back Images from v0.5.0 to v0.4.50
INFO: Rolling back Homebrew from v0.1.17 to v0.1.16
INFO: Rolling back FixedPointNumbers from v0.1.0 to v0.0.12
INFO: Rolling back Colors from v0.6.0 to v0.5.4
INFO: Rolling back ColorVectorSpace from v0.1.0 to v0.0.5
INFO: Rolling back ColorTypes from v0.2.0 to v0.1.7
INFO: Rolling back install of Clustering
INFO: Rolling back Calculus from v0.1.14 to v0.1.13
ERROR: failed process: Process(`git --work-tree=/Users/ken/.julia/v0.4/Plots --git-dir=/Users/ken/.julia/v0.4/Plots/.git checkout -q 894928d7ca2f14c4bf5aa4fe97b581736f4264c1`, ProcessExited(128)) [128]
 in pipeline_error at process.jl:555
 in run at process.jl:531
 in update at pkg/write.jl:44
 in resolve at /Applications/Julia-0.4.0.app/Contents/Resources/julia/lib/julia/sys.dylib
 in update at /Applications/Julia-0.4.0.app/Contents/Resources/julia/lib/julia/sys.dylib
 in anonymous at pkg/dir.jl:31
 in cd at file.jl:22
 in cd at pkg/dir.jl:31
 in update at /Applications/Julia-0.4.0.app/Contents/Resources/julia/lib/julia/sys.dylib

```

Then, running `git status` gives me:

```
$git status
HEAD detached at 1115ff8
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   .gitignore
	modified:   .travis.yml
	modified:   README.md
	modified:   docs/example_generation.jl
	modified:   docs/gadfly_examples.md
	modified:   docs/immerse_examples.md
	modified:   docs/pyplot_examples.md
	modified:   docs/qwt_examples.md
	modified:   docs/readme_template.md
	modified:   docs/unicodeplots_examples.md
	modified:   docs/winston_examples.md

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	.gitattributes
	examples/animations.ipynb
	examples/facets.ipynb
	examples/meetup/

no changes added to commit (use "git add" and/or "git commit -a")
```

Now, resetting gives me:
```
ken$ git reset --hard
HEAD is now at 1115ff8 working on contours
```
but still the `Pkg.update()` shows the same error.

**Comment by Ken-B** [comment]
Hmm, I just noticed Plots.jl was not in my required packages, but in my added packages:
```
julia> Pkg.status()
21 required packages:
 - Atom                          0.1.1+             master
 - Cairo                         0.2.31
 - Compose                       0.3.18+            master
 - Escher                        0.2.1
 - ForwardDiff                   0.1.2+             master
 - Gadfly                        0.3.18+            master
 - IJulia                        1.1.7
 - Images                        0.4.50
 - Interact                      0.2.1
 - Jewel                         1.0.7+             master
 - Mocha                         0.1.0
 - NullableArrays                0.0.1
 - ProfileView                   0.1.1
 - PyPlot                        2.1.1
 - QuantEcon                     0.2.1
 - Reactive                      0.2.4
 - SimJulia                      0.3.9
 - Sobol                         0.1.2-             SobolJulia
 - SymPy                         0.2.30+            master
 - TestImages                    0.0.8
 - Winston                       0.11.13
100 additional packages:
 - ArgParse                      0.3.0
 - ArrayViews                    0.6.4
 - AutoHashEquals                0.0.5
 - Benchmark                     0.1.0
 - Benchmarks                    0.0.0-             master (unregistered)
 - BinDeps                       0.3.18
 - Blosc                         0.1.4
 - Calculus                      0.1.13
 - CodeTools                     0.1.0
 - Codecs                        0.1.5
 - ColorTypes                    0.1.7
 - ColorVectorSpace              0.0.5
 - Colors                        0.5.4
 - Compat                        0.7.7
 - Conda                         0.1.7
 - Contour                       0.0.8
 - DSP                           0.0.11
 - DataArrays                    0.2.19
 - DataFrames                    0.6.10
 - DataStructures                0.3.13
 - Dates                         0.4.4
 - Debug                         0.1.6
 - Distances                     0.2.1
 - Distributions                 0.8.7
 - Docile                        0.5.19
 - DualNumbers                   0.1.5
 - Electron                      0.0.0-             master (unregistered)
 - FactCheck                     0.4.1
 - FastAnonymous                 0.3.2
 - FileIO                        0.0.3              master
 - FixedPointNumbers             0.0.12
 - Formatting                    0.1.4
 - FunctionalCollections         0.1.3
 - GZip                          0.2.18
 - Graphics                      0.1.3
 - Grid                          0.4.0
 - Gtk                           0.9.2
 - GtkUtilities                  0.0.6
 - HDF5                          0.5.6
 - Hexagons                      0.0.4
 - Hiccup                        0.0.1
 - Homebrew                      0.1.16
 - HttpCommon                    0.2.4
 - HttpParser                    0.1.1
 - HttpServer                    0.1.4
 - ImageIO                       0.0.0-             master (unregistered)
 - ImageMagick                   0.1.0              master
 - ImmutableArrays               0.0.11
 - IniFile                       0.2.4
 - IterationManagers             0.0.1
 - Iterators                     0.1.9
 - JLD                           0.5.5
 - JSON                          0.5.0
 - JuliaParser                   0.6.3+             master
 - KernelDensity                 0.1.2
 - LNR                           0.0.1
 - LaTeXStrings                  0.1.6
 - Lazy                          0.10.1
 - LeafAreaIndex                 0.0.0-             v0.4 (unregistered)
 - Lexicon                       0.1.16
 - LightGraphs                   0.3.4
 - LightXML                      0.2.1
 - Loess                         0.0.5
 - Logging                       0.1.1
 - LsqFit                        0.0.2
 - MAT                           0.2.14
 - MacroTools                    0.2.0
 - Markdown                      0.3.0
 - MbedTLS                       0.1.4
 - Media                         0.1.1
 - Memoize                       0.0.0
 - MinFinder                     0.0.0-             master (unregistered)
 - Mux                           0.1.1
 - NaNMath                       0.1.1
 - Nettle                        0.2.0              master
 - Optim                         0.4.4
 - PDMats                        0.3.6
 - ParserCombinator              1.7.1
 - Patchwork                     0.1.9
 - Plots                         0.4.0
 - Polynomials                   0.0.4
 - PyCall                        1.1.2
 - Reexport                      0.0.3
 - Requires                      0.2.0
 - Rutherford                    0.0.0-             master (unregistered)
 - SHA                           0.1.2
 - SIUnits                       0.0.5
 - Showoff                       0.0.6
 - SortingAlgorithms             0.0.6
 - StatsBase                     0.7.4
 - StatsFuns                     0.1.4
 - TexExtensions                 0.0.2
 - TextWrap                      0.1.5
 - Tk                            0.3.6
 - URIParser                     0.1.1
 - WebSockets                    0.1.1
 - WoodburyMatrices              0.1.2
 - ZMQ                           0.3.0
 - ZipFile                       0.2.5
 - Zlib                          0.1.12
```

So `Pkg.add("Plots")` and then `Pkg.update()` solved it for me. Seems like another package was somehow requiring a particular version of Plots that would not update cleanly? Very strange. 

@StefanKarpinski A bit worrying that this broke the Pkg.update for all packages. Let me know if I can do something to trace the issue down further or replicate.

**Comment by tbreloff** [comment]
Yes this is a big problem... I was trying out git-lfs as a possible solution for eliminating image files from the repo, but that was clearly a failed experiment.  I'm going to re-tag a new version that hopefully fixes this.  I'm actually surprised that an add/update solved your problem.

Can you verify if the latest master (`Pkg.checkout("Plots")`) works for you?

**Comment by Ken-B** [comment]
It seems the Plots package became dirty again.

```
julia> Pkg.checkout("Plots")
INFO: Checking out Plots master...
ERROR: Plots is dirty, bailing
 in error at /Applications/Julia-0.4.0.app/Contents/Resources/julia/lib/julia/sys.dylib
 in anonymous at pkg/entry.jl:191
 in transact at pkg/git.jl:97
 in _checkout at pkg/entry.jl:190
 in checkout at pkg/entry.jl:205
 in anonymous at pkg/dir.jl:31
 in cd at file.jl:22
 in cd at pkg/dir.jl:31
 in checkout at pkg.jl:37
```
And indeed, with `Pkg.status()` I get:
```
 - Plots                         0.4.0              1115ff83 (dirty)
```

and in the Plots directory:
```
ken$ git status
HEAD detached at 1115ff8
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	new file:   .gitattributes
	new file:   examples/animations.ipynb
	new file:   examples/facets.ipynb
	new file:   examples/meetup/cart-before-the-horse.jpg

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   .gitignore
	modified:   README.md
	modified:   docs/example_generation.jl
	modified:   docs/gadfly_examples.md
	modified:   docs/immerse_examples.md
	modified:   docs/pyplot_examples.md
	modified:   docs/qwt_examples.md
	modified:   docs/readme_template.md
	modified:   docs/unicodeplots_examples.md
	modified:   docs/winston_examples.md
```


I don't know much about git, but let me know what I can do to help.

**Comment by tbreloff** [comment]
I think the offending package is SymPy.  @jverzani is there a reason that you require `Plots 0.4.0`? (as opposed to just `Plots`)  It certainly shouldn't be causing these sorts of errors, but it still seems unwanted.

@Ken-B Can you fully delete the Plots directory from your filesystem, restart Julia, and try again?  I want to figure out if the SymPy requirement makes it impossible to have a different version.

**Comment by Ken-B** [comment]
Ok, so I did
```
julia> Pkg.rm("Plots")
INFO: No packages to install, update or remove
INFO: Package database updated
```
and then removed the `Plots` folder under `.julia/v0.4`.

However, I got:
```
julia> Pkg.resolve()
INFO: Installing Plots v0.4.1
git-lfs smudge 'examples/meetup/iheartplots.gif': git-lfs: command not found
error: cannot feed the input to external filter git-lfs smudge %f
error: external filter git-lfs smudge %f failed -1
error: external filter git-lfs smudge %f failed
fatal: examples/meetup/iheartplots.gif: smudge filter lfs failed
ERROR: failed process: Process(`git --work-tree=/Users/ken/.julia/v0.4/Plots --git-dir=/Users/ken/.julia/v0.4/Plots/.git checkout -q 894928d7ca2f14c4bf5aa4fe97b581736f4264c1`, ProcessExited(128)) [128]
 in pipeline_error at process.jl:555
 in run at process.jl:531
 in install at pkg/write.jl:38
 in resolve at /Applications/Julia-0.4.0.app/Contents/Resources/julia/lib/julia/sys.dylib
 in resolve at pkg/entry.jl:404
 in anonymous at pkg/dir.jl:31
 in cd at file.jl:22
 in cd at pkg/dir.jl:31
 in resolve at pkg.jl:46
```

**Comment by tbreloff** [comment]
That seems really strange.  Why is it trying to install Plots 0.4.1?

Anyways @jverzani I realized that you wanted a version which supports 0.3.  I'm going to try get one final 0.3 supported version with the cleaned out directory structure and tag it 0.4.2.  You should probably require `Plots 0.4` to get the latest bug fixes.

I'll post here after the new tags are published.

**Comment by tbreloff** [comment]
Ok a new 0.3-compatible version has been tagged: https://github.com/JuliaLang/METADATA.jl/pull/3908#event-449531944.  I hope this resolves some of the issues here, although it may require deleting some stuff and re-adding for @Ken-B .  Let me know what happens!

Also @jverzani... I think if you leave off the version number in your require file then 0.3 users will automatically get 0.4.2, even when I tag a 0.5.0. (at least... that's how I think it works...)

**Comment by jverzani** [comment]
Okay, I'll update the REQUIRE file. Your suspicions are right about why I
wanted that version of `Plots`, but what you suggest about picking up the
most recent version working with 0.3 makes sense. Sorry if it caused a big
hassle.

On Thu, Oct 29, 2015 at 2:09 PM, Tom Breloff <notifications@github.com>
wrote:

> Ok a new 0.3-compatible version has been tagged: JuliaLang/METADATA.jl#3908
> (comment)
> <https://github.com/JuliaLang/METADATA.jl/pull/3908#event-449531944>. I
> hope this resolves some of the issues here, although it may require
> deleting some stuff and re-adding for @Ken-B <https://github.com/Ken-B> .
> Let me know what happens!
>
> Also @jverzani <https://github.com/jverzani>... I think if you leave off
> the version number in your require file then 0.3 users will automatically
> get 0.4.2, even when I tag a 0.5.0. (at least... that's how I think it
> works...)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tbreloff/Plots.jl/issues/55#issuecomment-152269600>.
>



-- 
John Verzani
Chair, Department of Mathematics
College of Staten Island, CUNY
verzani@math.csi.cuny.edu

**Comment by tbreloff** [comment]
Not a problem... I'm happy to have problems show up so they can be fixed :)

**Comment by tbreloff** [comment]
I just checked and SymPy is hopefully the only package with this problem right now:

```
[tom@tomoffice METADATA]$ grep Plots */versions/*/requires
AverageShiftedHistograms/versions/0.2.0/requires:TextPlots
AverageShiftedHistograms/versions/0.2.1/requires:TextPlots
AverageShiftedHistograms/versions/0.2.2/requires:UnicodePlots
AverageShiftedHistograms/versions/0.3.0/requires:UnicodePlots
AverageShiftedHistograms/versions/0.3.0/requires:Plots
SymPy/versions/0.2.29/requires:Plots 0.4.0
SymPy/versions/0.2.30/requires:Plots 0.4.0
```

**Comment by jverzani** [comment]
the SymPy master has the change, I just need to push to METADATA. Is there
some way to test if this change helps at all?

On Thu, Oct 29, 2015 at 3:46 PM, Tom Breloff <notifications@github.com>
wrote:

> I just checked and SymPy is hopefully the only package with this problem
> right now:
>
> [tom@tomoffice METADATA]$ grep Plots */versions/*/requires
> AverageShiftedHistograms/versions/0.2.0/requires:TextPlots
> AverageShiftedHistograms/versions/0.2.1/requires:TextPlots
> AverageShiftedHistograms/versions/0.2.2/requires:UnicodePlots
> AverageShiftedHistograms/versions/0.3.0/requires:UnicodePlots
> AverageShiftedHistograms/versions/0.3.0/requires:Plots
> SymPy/versions/0.2.29/requires:Plots 0.4.0
> SymPy/versions/0.2.30/requires:Plots 0.4.0
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tbreloff/Plots.jl/issues/55#issuecomment-152299180>.
>



-- 
John Verzani
Chair, Department of Mathematics
College of Staten Island, CUNY
verzani@math.csi.cuny.edu

**Comment by tbreloff** [comment]
@Ken-B Can you do:

```
Pkg.checkout("SymPy")
 < might need to quit, remove Plots directory, restart? >
Pkg.add("Plots")
Pkg.update()
```

and let us know if the issue is resolved?

**Comment by tbreloff** [comment]
Also... you may want to remove it from your `.cache` directory as well.

**Comment by Ken-B** [comment]
So I removed Plots folder from v0.4 and from .cache, and I got:
```
julia> Pkg.resolve()
INFO: Cloning cache of Plots from git://github.com/tbreloff/Plots.jl.git
INFO: Installing Plots v0.4.1
git-lfs smudge 'examples/meetup/iheartplots.gif': git-lfs: command not found
error: cannot feed the input to external filter git-lfs smudge %f
error: external filter git-lfs smudge %f failed -1
error: external filter git-lfs smudge %f failed
fatal: examples/meetup/iheartplots.gif: smudge filter lfs failed
ERROR: failed process: Process(`git --work-tree=/Users/ken/.julia/v0.4/Plots --git-dir=/Users/ken/.julia/v0.4/Plots/.git checkout -q 894928d7ca2f14c4bf5aa4fe97b581736f4264c1`, ProcessExited(128)) [128]
 in pipeline_error at process.jl:555
 in run at process.jl:531
 in install at pkg/write.jl:38
 in resolve at /Applications/Julia-0.4.0.app/Contents/Resources/julia/lib/julia/sys.dylib
 in resolve at pkg/entry.jl:404
 in anonymous at pkg/dir.jl:31
 in cd at file.jl:22
 in cd at pkg/dir.jl:31
 in resolve at pkg.jl:46

julia> Pkg.add("Plots")
INFO: No packages to install, update or remove
INFO: Package database updated
INFO: METADATA is out-of-date — you may not have the latest version of Plots
INFO: Use `Pkg.update()` to get the latest versions of your packages

julia> Pkg.update()
INFO: Updating METADATA...
INFO: Updating cache of CodeTools...
INFO: Updating cache of IJulia...
INFO: Updating cache of Atom...
INFO: Updating cache of QuantEcon...
INFO: Updating Rutherford...
INFO: Updating ForwardDiff...
INFO: Updating Atom...
INFO: Updating MinFinder...
INFO: Updating JuliaParser...
INFO: Updating SymPy...
INFO: Updating Compose...
INFO: Updating ImageIO...
INFO: Updating ImageMagick...
INFO: Updating Gadfly...
INFO: Updating FileIO...
INFO: Updating Benchmarks...
INFO: Updating Electron...
INFO: Updating Sobol...
INFO: Updating LeafAreaIndex...
INFO: Updating Jewel...
INFO: Updating Nettle...
INFO: Computing changes...
INFO: Upgrading CodeTools: v0.1.0 => v0.2.0
INFO: Upgrading IJulia: v1.1.7 => v1.1.8
INFO: Upgrading QuantEcon: v0.2.1 => v0.4.0
INFO: Removing Formatting v0.1.4
INFO: Removing IterationManagers v0.0.1
INFO: Removing MAT v0.2.14
INFO: Building Homebrew
remote: Counting objects: 273, done.
remote: Compressing objects: 100% (104/104), done.
remote: Total 273 (delta 219), reused 223 (delta 169), pack-reused 0
Receiving objects: 100% (273/273), 33.50 KiB | 0 bytes/s, done.
Resolving deltas: 100% (219/219), completed with 34 local objects.
From https://github.com/Homebrew/homebrew
   ea5233c..8fcb8d2  master     -> origin/master
HEAD is now at 8fcb8d2 v: fix description
HEAD is now at ce3d0c3 Merge pull request #78 from staticfloat/staging
INFO: Building Nettle
INFO: Building ZMQ
INFO: Building IJulia
INFO: Found Jupyter version 4.0.6: jupyter
Writing IJulia kernelspec to /Users/ken/.julia/v0.4/IJulia/deps/julia-0.4/kernel.json ...
Installing julia kernelspec julia-0.4
[InstallKernelSpec] Removing existing kernelspec in /Users/ken/Library/Jupyter/kernels/julia-0.4
[InstallKernelSpec] Installed kernelspec julia-0.4 in /Users/ken/Library/Jupyter/kernels/julia-0.4
INFO: Building LightXML

julia> Pkg.status()
22 required packages:
 - Atom                          0.2.0              master
 - Cairo                         0.2.31
 - Compose                       0.3.18+            master
 - Escher                        0.2.1
 - ForwardDiff                   0.1.2+             master
 - Gadfly                        0.3.18+            master
 - IJulia                        1.1.8
 - Images                        0.5.0
 - Interact                      0.2.1
 - Jewel                         1.0.7+             master
 - Mocha                         0.1.0
 - NullableArrays                0.0.2
 - Plots                         0.4.2              master (dirty)
 - ProfileView                   0.1.1
 - PyPlot                        2.1.1
 - QuantEcon                     0.4.0
 - Reactive                      0.2.4
 - SimJulia                      0.3.9
 - Sobol                         0.1.2-             SobolJulia
 - SymPy                         0.2.30+            master
 - TestImages                    0.1.0
 - Winston                       0.11.13
97 additional packages:
 - ArgParse                      0.3.0
 - ArrayViews                    0.6.4
 - AutoHashEquals                0.0.5
 - Benchmark                     0.1.0
 - Benchmarks                    0.0.0-             master (unregistered)
 - BinDeps                       0.3.18
 - Blosc                         0.1.4
 - Calculus                      0.1.14
 - Clustering                    0.5.0
 - CodeTools                     0.2.0
 - Codecs                        0.1.5
 - ColorTypes                    0.2.0
 - ColorVectorSpace              0.1.0
 - Colors                        0.6.0
 - Compat                        0.7.7
 - Conda                         0.1.7
 - Contour                       0.0.8
 - DSP                           0.0.11
 - DataArrays                    0.2.19
 - DataFrames                    0.6.10
 - DataStructures                0.3.13
 - Dates                         0.4.4
 - Debug                         0.1.6
 - Distances                     0.2.1
 - Distributions                 0.8.7
 - Docile                        0.5.19
 - DualNumbers                   0.1.5
 - Electron                      0.0.0-             master (unregistered)
 - FactCheck                     0.4.1
 - FastAnonymous                 0.3.2
 - FileIO                        0.0.3+             master
 - FixedPointNumbers             0.1.0
 - FunctionalCollections         0.1.3
 - GZip                          0.2.18
 - Graphics                      0.1.3
 - Grid                          0.4.0
 - Gtk                           0.9.2
 - GtkUtilities                  0.0.6
 - HDF5                          0.5.6
 - Hexagons                      0.0.4
 - Hiccup                        0.0.1
 - Homebrew                      0.1.17
 - HttpCommon                    0.2.4
 - HttpParser                    0.1.1
 - HttpServer                    0.1.4
 - ImageIO                       0.0.0-             master (unregistered)
 - ImageMagick                   0.1.0              master
 - ImmutableArrays               0.0.11
 - IniFile                       0.2.4
 - Iterators                     0.1.9
 - JLD                           0.5.6
 - JSON                          0.5.0
 - JuliaParser                   0.6.3+             master
 - KernelDensity                 0.1.2
 - LNR                           0.0.1
 - LaTeXStrings                  0.1.6
 - Lazy                          0.10.1
 - LeafAreaIndex                 0.0.0-             master (unregistered)
 - LightGraphs                   0.4.0
 - LightXML                      0.2.1
 - Loess                         0.0.5
 - Logging                       0.2.0
 - LsqFit                        0.0.2
 - MacroTools                    0.2.0
 - Markdown                      0.3.0
 - MbedTLS                       0.1.4
 - Measures                      0.0.1
 - Media                         0.1.1
 - Memoize                       0.0.1
 - MinFinder                     0.0.0-             master (unregistered)
 - Mux                           0.1.1
 - NaNMath                       0.1.1
 - Nettle                        0.2.0              master
 - Optim                         0.4.4
 - PDMats                        0.3.6
 - ParserCombinator              1.7.1
 - Patchwork                     0.1.9
 - Polynomials                   0.0.4
 - PyCall                        1.1.2
 - Reexport                      0.0.3
 - Requires                      0.2.1
 - Rutherford                    0.0.0-             master (unregistered)
 - SHA                           0.1.2
 - SIUnits                       0.0.6
 - Showoff                       0.0.6
 - SortingAlgorithms             0.0.6
 - StatsBase                     0.7.4
 - StatsFuns                     0.2.0
 - TexExtensions                 0.0.3
 - TextWrap                      0.1.5
 - Tk                            0.3.6
 - URIParser                     0.1.1
 - WebSockets                    0.1.1
 - WoodburyMatrices              0.1.2
 - ZMQ                           0.3.0
 - ZipFile                       0.2.5
 - Zlib                          0.1.12
```

It installed Plots 0.4.2, but the repo is dirty, so I deleted the Plots directory again and now it seems to work:
```
julia> Pkg.update()
INFO: Updating METADATA...
INFO: Updating Rutherford...
INFO: Updating ForwardDiff...
INFO: Updating Atom...
INFO: Updating MinFinder...
INFO: Updating JuliaParser...
INFO: Updating SymPy...
INFO: Updating Compose...
INFO: Updating ImageIO...
INFO: Updating ImageMagick...
INFO: Updating Gadfly...
INFO: Updating FileIO...
INFO: Updating Benchmarks...
INFO: Updating Electron...
INFO: Updating Sobol...
INFO: Updating LeafAreaIndex...
INFO: Updating Jewel...
INFO: Updating Nettle...
INFO: Computing changes...
INFO: Installing Plots v0.4.2

```
I had the same issue on Windows, where deleting the Plots directory and then `Pkg.update()` solved it.

Thank you for the quick reaction and solution!

**Comment by tbreloff** [comment]
Ok glad you finally got it working!  Thanks for sticking with it and posting your results.

For future reference, the `Pkg.resolve` call actually re-dirtied your repo, which is why you had to go through the extra steps.  The order matters:

```
Pkg.rm("Plots")  # removes it from .julia/v0.4/REQUIRE
Pkg.checkout("SymPy")  # gets the new SymPy/REQUIRE that won't screw things up

< quit, delete Plots from main directory and the cache, then restart julia >

Pkg.update()   # gets the latest METADATA, which says that Plots should be 0.4.2, then installs it
Pkg.add("Plots")  # this should be unnecessary since SymPy requires it...
```

This process is getting an overhaul from the core julia developers, so that hopefully someday package dependencies aren't so fragile.  For now we need to live with the occasional confusion.

@jverzani Thanks again for the require update.  Just be aware that others might run into this issue until you tag a new version of SymPy.

**Comment by lewisl** [comment]
I had indeed deleted my entire packages subdirectory.  Installing again didn't make it dirty.  using Plots make it dirty.  But, it is fixed--as you surely know--in 0.4.2.  Thanks for all the prompt work in tracking it down.
